tests/collect_in.rs

```diff
@@ -1,28 +1,24 @@
 #![cfg(feature = "collections")]
+
+mod quickcheck;
```

Comment:
Instead of making every test program recompile the `quickcheck` module, let's reorganize the tests like

```
bumpalo/
  - tests/
    - all/
      - main.rs
      - quickcheck.rs
      - etc...
```

and `main.rs` has everything else as submodules (maybe `cfg`ed on or off depending what features are enabled and all that).

This way, we only compile the `quickcheck` module once, and don't need to have many `mod quickcheck` declarations.

Reply:
I don't see how you would avoid these "recompilations" anyway, if we keep tests separated inside of multiple files, since you need to `mod something` for each of these files. I don't think joining file would be a good idea too, i just tried to do what [the rust book says](https://doc.rust-lang.org/book/ch11-03-test-organization.html#submodules-in-integration-tests). I could take the `common/mod.rs` advice tho...

Reply:
Each `.rs` file under `tests/` is compiled as its own crate (including the module without any tests!). By having a module that is used by all of them, it is recompiled N times, once for each crate. With the layout I proposed, there is only one crate, and the module is therefore just compiled once. It might not be the hugest deal at the moment, but it sets us up for the growth of these tests in the future.

Reply:
> Each `.rs` file under `tests/` is compiled as its own crate (including the module without any tests!).

Yes, and it doesn't recurse into a directory. By moving `tests/quickcheck.rs` into `tests/quickcheck/mod.rs` in [0f538b1](https://github.com/fitzgen/bumpalo/pull/168/commits/0f538b11af7f25719585e11c005246f32dd0b388) i was able to delete this empty `quickcheck` test.

Keep in mind this tests structure is recommended by the rust book and, i believe, should be considered standard. Until this code works fine If you see an improvement to be made by changing tests structure, then i would ask you to handle it via a separate PR, since it would be outside of the scope of this PR. And one more thing, i would recommend not to do that restructure unless you detect a significant improvement in build time, disk usage, or something else.

Reply:
Also recompilations like these are common in C++ world with header-only libraries there.

Reply:
The C++ world is really a non sequitur here. Doesn't have any relevance.

The `tests/all/main.rs` structure is a common idiom:

* https://github.com/bytecodealliance/wasmtime/blob/main/tests/all/main.rs
* https://github.com/rustwasm/wasm-bindgen/blob/main/tests/wasm/main.rs
* https://github.com/rust-fuzz/cargo-fuzz/blob/master/tests/tests/main.rs
* it is even used in `cargo` itself: https://github.com/rust-lang/cargo/blob/master/tests/testsuite/main.rs

but like. man. I'm tired. we've been going back and forth on this for too long. I don't have the energy to keep this up. it's my repo. just structure it the way I prefer. thanks.

Reply:
Implemented in 161b65009096b18d20077cbb3e9fa09464e5e611